main.tf

```diff
@@ -1,16 +1,153 @@
+locals {
+  platforms = {
+    amazon_linux_2 = "AMAZON_LINUX_2"
+    amazon_linux   = "AMAZON_LINUX"
+    centos         = "CENTOS"
+    oracle         = "ORACLE_LINUX"
+    suse           = "SUSE"
+    windows        = "WINDOWS"
+    debian         = "DEBIAN"
+    ubuntu         = "UBUNTU"
+    rhel           = "REDHAT_ENTERPRISE_LINUX"
+    macos          = "MACOS"
+  }
+}
 
-module "tags" {
-  source  = "rhythmictech/tags/terraform"
-  version = "~> 1.1.0"
+data "aws_iam_policy_document" "ssm_maintenance_window" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type = "Service"
+      identifiers = [
+        "ec2.amazonaws.com",
+        "ssm.amazonaws.com"
+      ]
+    }
+  }
+}
 
-  enforce_case = "UPPER"
-  names        = [var.name]
-  tags         = var.tags
+resource "aws_iam_role" "ssm_maintenance_window" {
+  name_prefix        = var.name
+  assume_role_policy = data.aws_iam_policy_document.ssm_maintenance_window.json
+  path               = "/system/"
+  tags               = var.tags
 }
 
-locals {
-  # tflint-ignore: terraform_unused_declarations
-  name = module.tags.name
-  # tflint-ignore: terraform_unused_declarations
-  tags = module.tags.tags_no_name
+resource "aws_iam_role_policy_attachment" "role_attach_ssm_mw" {
+  role       = aws_iam_role.ssm_maintenance_window.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonSSMMaintenanceWindowRole"
+}
+
+data "aws_ssm_patch_baseline" "this" {
+  for_each         = local.platforms
+  owner            = "AWS"
+  name_prefix      = "AWS-"
+  operating_system = each.value
+  default_baseline = true
+}
+
+resource "aws_ssm_patch_group" "patchgroup" {
+  for_each    = local.platforms
+  baseline_id = data.aws_ssm_patch_baseline.this[each.key].id
+  patch_group = each.value
+}
+
+resource "aws_ssm_maintenance_window" "scan" {
+  name              = "scan-${var.name}"
+  cutoff            = 1
+  description       = "Maintenance window for applying patches"
+  duration          = 4
+  schedule          = "cron(0 06 * * ? *)"
+  schedule_timezone = "America/New_York"
+  tags              = var.tags
+}
+
+resource "aws_ssm_maintenance_window" "install" {
+  name              = "install-${var.name}"
+  cutoff            = 1
+  description       = "Maintenance window for applying patches"
+  duration          = 3
+  schedule          = "cron(0 22 * * ? *)"
+  schedule_timezone = "America/New_York"
```

Comment:
If we're going to hardcode this I think we should go with UTC

Reply:
I should just parameterize this. Do you think UTC default or no default?

Reply:
UTC default makes sense to me